bridge/telegram/handlers.go

```diff
@@ -512,6 +512,41 @@ func (b *Btelegram) handleEdit(msg *config.Message, chatid int64) (string, error
 // handleUploadFile handles native upload of files
 func (b *Btelegram) handleUploadFile(msg *config.Message, chatid int64, threadid int, parentID int) (string, error) {
 	var media []interface{}
+	equal := true
+	first := true
+	var prev string
+
+	for _, f := range msg.Extra["file"] {
+		fi := f.(config.FileInfo)
+		var ftype string
+
+		switch filepath.Ext(fi.Name) {
+		case ".jpg", ".jpe", ".png":
+			ftype = "image"
+		case ".mp4", ".m4v":
+			ftype = "video"
+		case ".mp3", ".oga", ".ogg", ".opus", ".flac":
+			ftype = "audio"
+		default:
+			ftype = "document"
+		}
+
+		if ftype == "document" {
+			equal = false
+			break
+		}
+
+		if first {
+			prev = ftype
+			equal = true
+		} else {
+			if prev != ftype {
+				equal = false
```

Comment:
Loop could be exited here already

Reply:
I fixed the condition that prevents entering to this place.

Reply:
Oh, you meant different. You wnated to say "You can break the loop here to don't waste resouces", ye. Done that.